gohypergiant/hyperdrive
# extract hyperpack onnx model (automl flavor)

## Summary

Given a hyperpack, 
- unzip the hyperpack onnx model saved therein
- for an automl model, convert the underlying onnx model to a pytorch model 

_Describe your changes._
- added `UnpackHyperpack` class to `utilities.py` in `hyperpackage` directory
- added `automl` to `hyperpackage/flavor/` directory for code to unpack automl models 

## Testing
- checkout this branch
- install hyperpackage with the following pip installation instruction:
    - `%pip install git+https://github.com/gohypergiant/hyperdrive.git@MLSDK472-extract-hp-onnxmodel#egg=hyperpackage\&subdirectory=hyperpackage` in a Jupyter notebook OR
    - `pip install git+https://github.com/gohypergiant/hyperdrive.git@MLSDK472-extract-hp-onnxmodel#egg=hyperpackage\&subdirectory=hyperpackage` from the command line

- check to make sure import hypackage works: 
```
from hyperpackage.utilities import UnpackHyperpack
from hyperpackage.flavors.automl import unpack_hp_univariate_ts as unpack_automl
```

- to test `UnpackHyperpack` class: 
(replace the directory `"./"` with the directory in which an unzipped hyperpack is located
```
unpacker = UnpackHyperpack('./')
unpacker.trial_names # for a list of trial names
# should return a Python list of trial names (will only be one since currently AutoML only saves the best trial) 
unpacker.onnx_filepaths # for a list of file paths (there should also only be one)
```

- to test `UnpackAutoMLHyperpack` class:
```
automl_unpacker = unpack_automl.UnpackAutoMLHyperpack('./', verbose=True)
automlts_onnx_model = automl_unpacker.load_automl_onnx_model()
automl_unpacker.onnx_models
automlts_torch_model = automl_unpacker.convert_onnx_to_automl_torch()
```

_Describe steps to follow to verify the changes that were made._

_If this ticket is a low-fi ticket specify what if anything is testable. A
"not testable" is perfectly acceptable in this case._

_Point out any previously tested functionality that may have been changed
and need another look._

## Notes

_Optional_

_Extra details about the PR that don't fit elsewhere such as dependencies
on another branch, a known issue that needs extra attention, or links._

## Screenshots

<img width="1117" alt="image" src="https://user-images.githubusercontent.com/32135867/191132891-afbdc7f2-ef79-4ddd-93cd-7a9be5b83cc8.png">
## Ticket(s)

- [MLSDK-472](https://gohypergiant.atlassian.net/browse/MLSDK-472)
